### PR TITLE
Optimize real-time functions

### PR DESCRIPTION
**Pull request by yxlllc**
1. Fcpe model cannot use directml inference, so directml mode will switch to cpu
2. Remove weight norm to improve model inference speed
3. Improve the cross-fade algorithm and add a new phase vocoder algorithm (optional)
4. Support setting the device sampling rate so that other audio APIs such as WASAPI can be used
5. Other code and inference speed optimizations

### CODE REVIEW

**Review by fumiama — DISMISSED** [review]

File: gui_v1.py [hunk 1]
```diff
@@ -265,35 +309,35 @@ def launcher(self):
...
-                                    default=data.get("rmvpe", "") == True,
+                                    default=data.get("rmvpe", False),
                                     enable_events=True,
                                 ),
                                 sg.Radio(
                                     "fcpe",
                                     "f0method",
                                     key="fcpe",
-                                    default=data.get("fcpe", "") == True,
+                                    default=data.get("fcpe", True),
```
Comment (Tps-F) [inline gui_v1.py:340]: I think this value is false.

**Review by Tps-F — COMMENTED** [review]
Test later and Approve if there are no problems.

File: gui_v1.py (hunk 1 above)
Comment (fumiama) [reply gui_v1.py:340]: Sure.

Comment (Tps-F) [reply gui_v1.py:340]: @yxlllc Please check here

**Review by Tps-F — CHANGES_REQUESTED** [review]

File: gui_v1.py (hunk 1 above)
Comment (yxlllc) [reply gui_v1.py:340]: Fcpe has a very strong performance and is much faster than rmvpe so it can be the default

Comment (fumiama) [reply gui_v1.py:340]: Well, then no problem since that's your purpose.

**Review by fumiama — DISMISSED** [review]

**Review by fumiama — APPROVED** [review]

**Pull request merged**